lib/util.js

```diff
@@ -58,72 +61,14 @@ export function privateKeyDerEncode({privateKeyBytes, seedBytes}) {
     p = seedBytes;
   } else {
     // extract the first 32 bytes of the 64 byte private key representation
-    p = Buffer.from(privateKeyBytes.buffer, privateKeyBytes.byteOffset, 32);
+    p = privateKeyBytes.slice(0, 32);
   }
-  const keyBuffer = new ByteBuffer(p);
-
-  const asn1Key = asn1.create(
-    asn1.UNIVERSAL,
-    asn1.Type.OCTETSTRING,
-    false,
-    keyBuffer.getBytes()
-  );
-
-  const a = asn1.create(
-    asn1.Class.UNIVERSAL,
-    asn1.Type.SEQUENCE,
-    true, [
-      asn1.create(
-        asn1.Class.UNIVERSAL, asn1.Type.INTEGER, false,
-        asn1.integerToDer(0).getBytes()),
-      // privateKeyAlgorithm
-      asn1.create(asn1.Class.UNIVERSAL, asn1.Type.SEQUENCE, true, [
-        asn1.create(
-          asn1.Class.UNIVERSAL,
-          asn1.Type.OID,
-          false,
-          asn1.oidToDer(oids.EdDSA25519).getBytes()
-        ),
-      ]),
-      // private key
-      asn1.create(
-        asn1.Class.UNIVERSAL, asn1.Type.OCTETSTRING, false,
-        asn1.toDer(asn1Key).getBytes()),
-    ]
-  );
-
-  const privateKeyDer = asn1.toDer(a);
-  return Buffer.from(privateKeyDer.getBytes(), 'binary');
+  return Buffer.concat([DER_PRIVATE_KEY_PREFIX, p]);
 }
 
 export function publicKeyDerEncode({publicKeyBytes}) {
   if(!(Buffer.isBuffer(publicKeyBytes) && publicKeyBytes.length === 32)) {
```

Comment:
This should be a bit more liberal without causing trouble:

```suggestion
  if(!(publicKeyBytes instanceof Uint8Array && publicKeyBytes.length === 32)) {
```

Reply:
I'm assuming `privateKeyBytes` and `seedBytes` can also be `instanceof Uint8Array`?